numpy/lib/function_base.py

```diff
@@ -944,11 +944,15 @@ def piecewise(x, condlist, funclist, *args, **kw):
             condlist = condlist.T
     if n == n2 - 1:  # compute the "otherwise" condition.
         totlist = np.logical_or.reduce(condlist, axis=0)
-        condlist = np.vstack([condlist, ~totlist])
+        try:
```

Comment:
Can you do this without using `try...except`? Using exception handling for flow  control is kind of ugly. Also, there should be an explanation of why the alternative is correct.